hack/bulk-ingest-explorer-files.py

```diff
@@ -0,0 +1,131 @@
+import google.auth
+import re
+import sys
+from collections import namedtuple
+from google.auth.transport.requests import AuthorizedSession
+from polling import poll
+from requests.exceptions import HTTPError
+from time import perf_counter
+
+
+# This script bulk-ingests files from a google bucket, with the following arguments:
+# - A profileID (associated with the billing account)
+# - The environment to use (dev/prod), will default to dev
+# - The google bucket path of control file (a json file which lists source and target paths)
+# - An optional dataset id (if none is provided, a temporary dataset will be used and deleted afterwards)
+
+# read in arguments
+profile_id = sys.argv[1]
+is_production = (sys.argv[2] == 'prod') # default to dev
+control_file_path = sys.argv[3]
+dataset_id = sys.argv[4] if sys.argv[4:] else None # optional
+
+credentials, project = google.auth.default(scopes=['openid', 'email', 'profile'])
+authed_session = AuthorizedSession(credentials)
+Counts = namedtuple('Counts', ['succeeded', 'failed', 'not_tried'])
+jade_base_url = 'https://jade-terra.datarepo-prod.broadinstitute.org/' if is_production \
+    else 'https://jade.datarepo-dev.broadinstitute.org/'
```

Comment:
I was on the fence about the best way to take the environment (dev/prod) as an argument, and then get the URL to use based on that. Let me know if you have any suggestions for how this could be improved.

Reply:
I tried looking in other Python scripts in our projects and it looks like we use:
`base_url = os.environ["API_URL"]`
Was looking at this script:
https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/scripts/poll-ingest-job.py
Looks like you might be able to do the same with some of the other arguments, I'm just forgetting where we  would make the change to set that API_URL

Reply:
Good point. I think for those, it's being set as an env variable when the scripts are called (like [here](https://github.com/DataBiosphere/clinvar-ingest/blob/b6e5b995fae6de5583b6ac722375a5a68eca9392/orchestration/templates/process-xml-release.yaml#L208)). 

But I agree that getting the url itself in as the argument (instead of "dev" or "prod") would make more sense.

Reply:
Just updated it! So now it looks something like this to run the script on the command line:
```python3 bulk-ingest-explorer-files.py <profile_id> <base_url> <control_file_path> <dataset_id>```

Reply:
nice, that looks god. so this script is just being called manually for the time being? oh actually I think I see some tickets to update the code to call this..

Reply:
Yeah, I have just been running it manually. In theory, this ingest will only be run once, so we don't need to set it up with too much infrastructure. 

My understanding is that a bunch of the encode files were copied over to a Broad-owned Google bucket, when they were building the "Encode Explorer" (when monster had just become a team). The Encode Explorer isn't updated anymore so we would only need to copy the files out of it once. But, it's easier (and maybe cheaper?) to move the big set of files from the encode explorer to jade, than to move them from the Encode site itself to Jade (because GCS -> GCS is easier than AWS -> GCS). 

So the next ticket is to do some performance testing and then do the _single run_ of this script in production that will copy the ~50,000 files or (about 95 TB).